make/build_test.mk

```diff
@@ -39,14 +40,14 @@ $($(T)_TEST_OBJ_DIR)/%.o: $($(T)_TEST_ROOT)/%.c | $(T) $(dir $($(T)_TEST_OBJ))
 # Compile the test runners
 $($(T)_TEST_OBJ_DIR)/%.o: $($(T)_GEN_DIR)/%.c | $(T) $(dir $($(T)_TEST_RUNNERS_OBJ))
 	@echo "T: $(notdir $<) -> $(notdir $@)"
-	@$(CC) -MD -MP -w -c -o $@ $< $($(firstword $|)_CFLAGS) $(addprefix -I,$(INC_DIRS))
+	@$(CC) -MD -MP -w -c -o $@ $($(T)_GEN_DIR)/$(notdir $<) $($(firstword $|)_CFLAGS) $(addprefix -I,$(INC_DIRS))
 
 # Build each test - only include the test's runner and unit tests.
 $($(T)_TESTS): $($(T)_TEST_BIN_DIR)/%_runner$(PLATFORM_EXT): \
                  $($(T)_TEST_OBJ_DIR)/%.o $($(T)_TEST_OBJ_DIR)/%_runner.o \
                  $(call dep_to_lib,$($(T)_TEST_DEPS)) | $($(T)_TEST_BIN_DIR)
 	@echo "Building test $(notdir $@) for $(PLATFORM)"
-	@$(CC) $(CFLAGS) -Wl,-Map=$|/$(notdir $(@:%$(PLATFORM_EXT)=%.map)) $^ -o $@ -L$(STATIC_LIB_DIR) \
+	@$(CC) $(CFLAGS) -Wl,-Map=$|/$(notdir $(@:%$(PLATFORM_EXT)=%.map)) $^ -o $(firstword $|)/$(notdir $@) -L$(STATIC_LIB_DIR) \
```

Comment:
I've been trying to get this to work for about 10 min. Are you referring to `DIRS` at the bottom of this file or `DIRS` in the main Makefile? Is there something I need to do aside from just adding the path to one of those because I am still getting errors from the ruby generators that the path doesn't exist even though I verified it was in DIRS.